application/common/components/MfaBackendWebAuthn.php

```diff
@@ -0,0 +1,187 @@
+<?php
+namespace common\components;
+
+use common\models\Mfa;
+use common\models\User;
+use GuzzleHttp\Exception\GuzzleException;
+use yii\base\Component;
+use yii\helpers\Json;
+use yii\web\NotFoundHttpException;
+use yii\web\ServerErrorHttpException;
+
+class MfaBackendWebAuthn extends Component implements MfaBackendInterface
+{
+    /**
+     * @var string
+     */
+    public string $apiBaseUrl;
+
+    /**
+     * @var string
+     */
+    public string $apiKey;
+
+    /**
+     * @var string
+     */
+    public string $apiSecret;
+
+    /**
+     * @var MfaApiClient
+     */
+    public MfaApiClient $client;
+
+    /**
+     * @var string
+     */
+    public string $appId;
+
+    /**
+     * @var string
+     */
+    public string $rpDisplayName;
+
+    /**
+     * @var string
+     */
+    public string $rpId;
+
+    public function init()
+    {
+        $this->client = new MfaApiClient($this->apiBaseUrl, $this->apiKey, $this->apiSecret);
+        parent::init();
+    }
+
+    /**
+     * Initialize a new WebAuthn registration
+     * @param int $userId The User ID
+     * @param string $rpOrigin The Replay Party Origin URL (with scheme, without port or path)
+     * @return array JSON decoded object to be passed to browser credential create API for WebAuthn dance
+     * @throws GuzzleException
+     */
+    public function regInit(int $userId, string $rpOrigin = ''): array
+    {
+        $user = User::findOne(['id' => $userId]);
+        if ($user == null) {
+            return [];
+        }
+
+        $headers = $this->getWebAuthnHeaders($user->username, $user->getDisplayName(), $rpOrigin);
+
+        return $this->client->webauthnCreateRegistration($headers);
+    }
+
+    /**
+     * Initialize authentication sequence
+     * @param int $mfaId The MFA ID
+     * @param string $rpOrigin The Replay Party Origin URL (with scheme, without port or path)
+     * @return array
+     * @throws NotFoundHttpException
+     * @throws GuzzleException
+     */
+    public function authInit(int $mfaId, string $rpOrigin = ''): array
```

Comment:
Would it work to save `rpOrigin` as an object variable so it doesn't have to be on every function signature?

Reply:
perhaps, but currently all the object variables are set via the component config